scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/S3ConnectorDownloader.java

```diff
@@ -0,0 +1,250 @@
+
+    private String scheme;
+
+    private String accessKey;
+
+    private String secretKey;
+
+    private boolean pathStyle = true;
+
+    private static final String S3_LOCAL_RELATIVE_PATH = "s3LocalRelativePath";
+
+    private static final String S3_URI = "s3Url";
+
+    /**
+     * @see JavaExecutable#init(Map)
+     */
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+
+        if (args.containsKey(S3_URI) && !args.get(S3_URI).toString().isEmpty()) {
+            s3Url = args.get(S3_URI).toString();
+            parseAmazonS3URI(s3Url);
+        } else {
+            throw new IllegalArgumentException("You have to specify a valid s3URI. Empty value is not allowed.");
+        }
+
+        if (args.containsKey(S3_LOCAL_RELATIVE_PATH) && !args.get(S3_LOCAL_RELATIVE_PATH).toString().isEmpty()) {
+            s3LocalRelativePath = args.get(S3_LOCAL_RELATIVE_PATH).toString();
+        } else {
+            //Default value is getLocalSpace() because it will always be writable and moreover can be used to transfer files to another data space (global, user)
+            s3LocalRelativePath = getLocalSpace();
+        }
+
+        accessKey = getThirdPartyCredential("S3_ACCESS_KEY");
+        secretKey = getThirdPartyCredential("S3_SECRET_KEY");
+        if (accessKey == null || secretKey == null) {
+            throw new IllegalArgumentException("You first need to add your s3 username and password (S3_ACCESS_KEY, S3_SECRET_KEY) to the third party credentials");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException {
+        List<String> filesRelativePathName = new ArrayList<>();
+
+        boolean s3PathIsPrefix = (keyName.lastIndexOf('/') == keyName.length() - 1);
+
+        File f = new File(s3LocalRelativePath);
+
+        // If the path already exists, print a warning.
+        if (f.exists()) {
+            getOut().println("The local path already exists");
+            if (s3PathIsPrefix) {
+                try {
+                    f.mkdir();
+                } catch (Exception e) {
+                    getOut().println("Couldn't create destination directory!");
+                    System.exit(1);
+                }
+            }
+        }
+        AmazonS3 amazonS3 = getS3Client(host, scheme);
+
+        // Assume that the path exists, do the download.
+        if (s3PathIsPrefix) {
+            downloadDir(bucketName, keyName, s3LocalRelativePath, false, amazonS3);
+            filesRelativePathName = SchedulerExamplesUtils.listDirectoryContents(f, new ArrayList<>());
+        } else {
+            s3LocalRelativePath = Paths.get(s3LocalRelativePath, Paths.get(s3Url).getFileName().toString()).toString();
+            downloadFile(bucketName, keyName, s3LocalRelativePath, false, amazonS3);
+            filesRelativePathName.add(s3LocalRelativePath);
```

Comment:
return Arrays.asList(s3LocalRelativePath)

Reply:
return Arrays.asList(s3LocalRelativePath); causes `java - Incompatible type with Arrays.asList()`
I replaced it with `return new ArrayList<String>() {{ add(s3LocalRelativePath);}};`